oscore-cf/src/main/java/org/eclipse/californium/oscore/CoapOSExceptionHandler.java

```diff
@@ -67,7 +68,9 @@ public static Response manageError(CoapOSException e, Request request) {
 					error.setType(tmp);
 					error.setPayload(e.getMessage());
 					error.getOptions().setMaxAge(0);
-
+					//Set MID of error response to match request
+					error.setMID(request.getMID());
+					
```

Comment:
Are you sure, that this is required?
According the javadoc of `Response.createResponse` that's set by the ReliabilityLayer.
Is that too late? Is there a unit test, which shows the defect without this?

Reply:
As I recall this was an issue found during our internal testing. I could create a unit test that tests this specifically.

Reply:
I have now added JUnit tests (OSCoreServerClientTest) that test a normal OSCORE request/response and also an OSCORE request where the response is a non-OSCORE error message. In the latter case the test will not work if the line above setting the MID for the error message is removed. This was added in commit ff646c0554141746d2078ee3aa996661cf99f250